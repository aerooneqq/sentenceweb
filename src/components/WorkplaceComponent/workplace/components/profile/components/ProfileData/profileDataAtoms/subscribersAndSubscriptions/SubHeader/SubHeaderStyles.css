#subHeader{ 
    width: 100%;
    height: 30px;
    background:#f0f0f0;

    display: flex;
    flex-direction: row;

    margin-top:-20px;
}

.subHeaderOption{
    height: 100%; 
    flex-grow: 1;

    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.subHeaderOption:nth-child(1){ 
    border-style: solid;
    border-width: 0;
    border-right-width: 0.05px;
    border-color: gray;
}

.subHeaderOption:nth-child(2){ 
    border-style: solid;
    border-width: 0;
    border-left-width: 0.05px;
    border-color: gray;
}

.subHeaderOption:hover .subHeaderOptionHighlighter{ 
    background: orangered;
}

.sunHeaderOptionTitleClicked{ 
    color: orangered !important;
}

.subHeaderOptionTitle{
    flex-grow: 1;

    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    cursor: default;

    font-family: Arial;
    color: gray;
}

.subHeaderOptionHighlighter{ 
    width: 100%;
    height: 2px;
    background: transparent;

    border-style: none;
    border-radius:  2px;
}