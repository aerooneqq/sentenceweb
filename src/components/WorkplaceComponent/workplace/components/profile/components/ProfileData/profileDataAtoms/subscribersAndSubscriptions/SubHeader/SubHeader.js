import React, {Component} from "react";

//Styles
import "./SubHeaderStyles.css";

/**
 * PROPS LIST:
 * 1) changeUserFriendMode
 */
export default class SubsHeader extends Component{ 
    constructor(props){ 
        super(props);

        this.changeFriendModeToSubscribers = this.changeFriendModeToSubscribers.bind(this);
        this.changeFriendModeToSubscriptions = this.changeFriendModeToSubscriptions.bind(this);
        this.changeFrinedsMode = this.changeFriendsMode.bind(this);
    }

    changeFriendsMode(newStatus) { 
        this.props.changeUserFriendsMode(newStatus);
    }

    changeFriendModeToSubscribers() { 
        this.changeFrinedsState("subscribers");

        document.getElementById("subscribersHeader").classList
            .toggle("subHeaderOptionTitleClicked");
        document.getElementById("subscriptionsHeader").classList
            .remove("subHeaderOptionTitleClicked");
    }

    changeFriendModeToSubscriptions() { 
        this.changeFrinedsState("subscriptions");

        document.getElementById("subscribersHeader").classList
            .remove("subHeaderOptionTitleClicked");
        document.getElementById("subscriptionsHeader").classList
            .toggle("subHeaderOptionTitleClicked");
    }

    render(){ 
        return( 
            <div id = "subHeader">
                <div id = "subscribers" 
                     className = "subHeaderOption"
                     onClick = {this.changeFriendModeToSubscribers}>
                    <div id = "subscribersHeader" className = "subHeaderOptionTitle">Subscribers</div>
                    <div className = "subHeaderOptionHighlighter" />
                </div>
                <div id = "subscriptions" 
                     className = "subHeaderOption"
                     onClick = {this.changeFriendModeToSubscriptions}>
                    <div id = "subscriptionsHeader" className = "subHeaderOptionTitle">Subscriptions</div>
                    <div className = "subHeaderOptionHighlighter" />
                </div>
            </div>
        );
    }
}