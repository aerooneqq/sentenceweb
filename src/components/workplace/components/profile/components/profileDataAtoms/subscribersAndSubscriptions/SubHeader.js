import React, {Component} from "react";

//Styles
import "./styles/SubHeaderStyles.css";

export default class SubsHeader extends Component{ 
    constructor(props){ 
        super(props);

        this.changeFriendModeToSubscribers = this.changeFriendModeToSubscribers.bind(this);
        this.changeFriendModeToSubscriptions = this.changeFriendModeToSubscriptions.bind(this);
    }

    changeFriendModeToSubscribers() { 
        this.props.changeUserFriendMode("subscribers");

        document.getElementById("subscribersHeader").classList
            .toggle("sunHeaderOptionTitleClicked");
        document.getElementById("subscriptionsHeader").classList
            .remove("sunHeaderOptionTitleClicked");
    }

    changeFriendModeToSubscriptions() { 
        this.props.changeUserFriendMode("subscriptions");

        document.getElementById("subscribersHeader").classList
            .remove("sunHeaderOptionTitleClicked");
        document.getElementById("subscriptionsHeader").classList
            .toggle("sunHeaderOptionTitleClicked");
    }

    render(){ 
        return( 
            <div id = "subHeader">
                <div id = "subscribers" 
                     className = "subHeaderOption"
                     onClick = {this.changeFriendModeToSubscribers}>
                    <div id = "subscribersHeader" className = "subHeaderOptionTitle">Subscribers</div>
                    <div className = "subHeaderOptionHighlighter" />
                </div>
                <div id = "subscriptions" 
                     className = "subHeaderOption"
                     onClick = {this.changeFriendModeToSubscriptions}>
                    <div id = "subscriptionsHeader" className = "subHeaderOptionTitle">Subscriptions</div>
                    <div className = "subHeaderOptionHighlighter" />
                </div>
            </div>
        );
    }
}