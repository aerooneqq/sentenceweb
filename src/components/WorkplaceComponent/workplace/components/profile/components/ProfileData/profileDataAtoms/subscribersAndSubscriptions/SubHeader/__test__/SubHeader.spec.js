import React from "react";
import {shallow} from "enzyme";
import {shallowToJson} from "enzyme-to-json";
import SubHeader from "../SubHeader";

const propsSpy = jest.fn(() => {});
const props = { 
    changeUserFriendsMode: propsSpy
}


describe("SubHeader", () => { 
    it("should render correctly", () => {
        expect(shallowToJson(shallow(<SubHeader />))).toMatchSnapshot();
    });

    it("should call changeFriendModeToSubscribers when clicked Subscribers are clicked", () => { 
        const spy = jest.spyOn(SubHeader.prototype, "changeFriendModeToSubscribers").mockImplementation(() => {});
        const subHeader = shallow(<SubHeader />);

        subHeader.find("div#subHeader").find("div#subscribers").simulate("click");
        expect(spy).toBeCalled();
    });

    it("should call changeFriendModeToSubscriptions method when Subscriptions are clicked", () => { 
        const spy = jest.spyOn(SubHeader.prototype, "changeFriendModeToSubscriptions").mockImplementation(() => {});
        const subHeader = shallow(<SubHeader />);

        subHeader.find("div#subHeader").find("div#subscriptions").simulate("click");
        expect(spy).toBeCalled();
    });

    it("should call changeUserFriendMode when changeFrinedsState is called", () => { 
        const subHeader = shallow(<SubHeader {...props}/>);

        subHeader.instance().changeFriendsMode("");

        expect(propsSpy).toBeCalled();
    });
});