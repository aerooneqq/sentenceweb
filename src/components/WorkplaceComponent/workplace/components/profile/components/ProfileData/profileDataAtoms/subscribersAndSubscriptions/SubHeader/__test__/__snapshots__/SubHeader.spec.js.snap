// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubHeader should render correctly 1`] = `
<div
  id="subHeader"
>
  <div
    className="subHeaderOption"
    id="subscribers"
    onClick={[Function]}
  >
    <div
      className="subHeaderOptionTitle"
      id="subscribersHeader"
    >
      Subscribers
    </div>
    <div
      className="subHeaderOptionHighlighter"
    />
  </div>
  <div
    className="subHeaderOption"
    id="subscriptions"
    onClick={[Function]}
  >
    <div
      className="subHeaderOptionTitle"
      id="subscriptionsHeader"
    >
      Subscriptions
    </div>
    <div
      className="subHeaderOptionHighlighter"
    />
  </div>
</div>
`;
